tests/sqlparser_postgres.rs

```diff
@@ -594,6 +594,13 @@ fn parse_alter_table_constraints_rename() {
     }
 }
 
+#[test]
+fn parse_alter_table_constraints_unique_nulls_distinct() {
+    pg_and_generic().verified_stmt("ALTER TABLE t ADD CONSTRAINT b UNIQUE NULLS NOT DISTINCT (c)");
```

Comment:
For one of the test cases can we include an assertion on the AST, like in [this style](https://github.com/apache/datafusion-sqlparser-rs/blob/a54331889874ca896100b6666720d6fcb43b6965/tests/sqlparser_postgres.rs#L882-L888)? just to sanity check where in the statement the nulls option ends up in the tree